newscoop/library/Newscoop/Entity/Repository/ArticleTopicRepository.php

```diff
@@ -89,4 +88,32 @@ public function getTheOccurrenceOfTheTopic($topicId)
 
         return $query;
     }
+
+    /**
+     * Count article topics by given criteria.
+     *
+     * @param array $criteria
+     *
+     * @return int
+     */
+    public function countBy(array $criteria = array())
+    {
+        $queryBuilder = $this->createQueryBuilder('t')
+            ->select('COUNT(t)');
+
+        foreach ($criteria as $property => $value) {
+            if (!is_array($value)) {
+                $queryBuilder->andWhere("t.$property = :$property");
+            }
+        }
+
+        $query = $queryBuilder->getQuery();
```

Comment:
you do this same loop twice - first to add where options and later to attach parameters values. it can be done in one. You can add aprameters to query builder, you don't need query for that.